editors/vscode/server/src/main.rs

```diff
@@ -236,20 +249,45 @@ impl LanguageServer for Backend {
 }
 
 impl Backend {
-    fn init(&self, root_uri: Option<Url>) -> Result<()> {
+    async fn init(&self, root_uri: Option<Url>) -> Result<()> {
         self.root_uri.set(root_uri).map_err(|err| {
             let message = match err {
                 SetError::AlreadyInitializedError(_) => "root uri already initialized".into(),
                 SetError::InitializingError(_) => "initializing error".into(),
             };
 
             Error { code: ErrorCode::ParseError, message, data: None }
-        })
+        })?;
+        let uri = self
+            .root_uri
+            .get()
+            .expect("The root uri should be initialized already")
+            .as_ref()
+            .expect("should get uri");
+        let mut builder = globset::GlobSetBuilder::new();
+        // Collecting all ignore files
+        builder.add(Glob::new("**/.eslintignore").unwrap());
+        builder.add(Glob::new("**/.gitignore").unwrap());
+
+        let ignore_file_glob_set = builder.build().unwrap();
+
+        let mut gitignore_builder = ignore::gitignore::GitignoreBuilder::new(uri.path());
+        let walk = ignore::WalkBuilder::new(uri.path()).ignore(true).hidden(false).git_global(false).build();
+        for item in walk.into_iter() {
+            if let Ok(entry) = item {
+                if ignore_file_glob_set.is_match(entry.path()) {
+                    gitignore_builder.add(entry.path());
+                }
+            }
+        }
+
+        *self.gitignore_glob.lock().await = gitignore_builder.build().ok();
```

Comment:
Let's move these new code to a function `init_file_ignore` into a function instead of dumping them here.